rfq/negotiator.go

```diff
@@ -293,6 +293,57 @@ func (n *Negotiator) HandleIncomingBuyRequest(
 	return nil
 }
 
+// HandleOutgoingSellOrder handles an outgoing sell order by constructing sell
+// requests and passing them to the outgoing messages channel. These requests
+// are sent to peers.
+func (n *Negotiator) HandleOutgoingSellOrder(order SellOrder) error {
+	// Query the price oracle for a reasonable ask price. We perform this
+	// query and response handling in a separate goroutine in case it is a
+	// remote service and takes a long time to respond.
+	n.Wg.Add(1)
+	go func() {
```

Comment:
`queryAskFromPriceOracle` is supposed to be the function for the oracle-asking logic. The rest of the code creates a new request and puts it on the outgoing channel.

This is where I think something like this might come in handy:
```golang
func execGoRoutine(cg *fn.ContextGuard, errChan chan<- error,
    exec func() error) error {

    // Ensure that the error channel has a buffer of at least 1.
    if cap(errChan) == 0 {
        return fmt.Errorf("error channel has no buffer")
    }

    cg.Wg.Add(1)
    go func() {
        defer cg.Wg.Done()

        err := exec()
        if err != nil {
            sendSuccess := fn.SendOrQuit(errChan, err, cg.Quit)
            if !sendSuccess {
                log.Warn("Failed to send error on error "+
                    "channel: %v", err)
            }
        }
    }()
    
    return nil
}
```

I'll see if i can refactor later. Is this nit blocking for you?

Reply:
nop, looks good!